youtube_dl/extractor/twitter.py

```diff
@@ -440,14 +456,17 @@ def _real_extract(self, url):
                 'tweet_mode': 'extended',
             })
 
-        title = description = status['full_text'].replace('\n', ' ')
-        # strip  'https -_t.co_BJYgOjSeGA' junk from filenames
-        title = re.sub(r'\s+(https?://[^ ]+)', '', title)
+        # Get metadata
+        description = status['full_text'].replace('\n', ' ')
         user = status.get('user') or {}
         uploader = user.get('name')
-        if uploader:
-            title = '%s - %s' % (uploader, title)
         uploader_id = user.get('screen_name')
+        timestamp = unified_timestamp(status.get('created_at'))
+
+        # Keep the file name short so it doesn't exceed filesystem limits
+        title = self._generic_title(url)
+        if uploader_id:
+            title = '@%s-%s' % (uploader_id, title)
```

Comment:
I'd rather use the previous title == description and just truncate it, or otherwise abbreviate it. Thus:
* pick a maximum length that will be reasonable to use in the default output template `%(title)s-%(id)s.%(ext)s`, say 40;
* pick an ellipsis string, say `[...]`;
* if the putative title resulting from old line 445 is longer than the maximum length, replace n characters either at the end or in the middle of the putative title by the ellipsis string, where n is the excess length of the putative title plus the length of the ellipsis string;
* if the putative title will have `uploader - ` prepended, add the length of `uploader` plus 3 to n.

Eg:
```py
        title = description = status['full_text'].replace('\n', ' ')
        # strip  'https -_t.co_BJYgOjSeGA' junk from filenames
        title = re.sub(r'\s+(https?://\S+)', '', title)
        # Get metadata
        user = status.get('user') or {}
        uploader = user.get('name')

        # Keep the file name short so it doesn't exceed filesystem limits

        def abbrev(field, size, where=None, ellipsis_str='[...]'):
            ellipsis_len = len(ellipsis_str)
            where = (size - ellipsis_len + 1) // 2 if where is None else (where - ellipsis_len)
            to_strip = len(field) - size + ellipsis_len
            if to_strip > 0:
                rest_len = len(field) - where - to_strip
                return field[:where] + ellipsis_str + (field[-rest_len:] if rest_len > 0 else '')
            return field

        MAX_TITLE_LEN = 40
        if uploader:
            # also limit length of uploader
            uploader_n = abbrev(uploader, 12, 12, '...')
            title = '%s - %s' % (uploader_n, abbrev(title, MAX_TITLE_LEN - len(uploader_n) - 3))
        else:
            title = abbrev(title, MAX_TITLE_LEN)
```

Reply:
Copy. I will make this change and let you know when it's done